### PR TITLE
feat: Add my orders page

### PR DESCRIPTION
**Pull request by yurii-ve**
UI:

Screenshot 2024-11-07 at 12 03 23

Screenshot 2024-11-07 at 12 02 43

### CODE REVIEW

File: app/routes/members-area.my-orders/route.module.scss [hunk 1]
```diff
@@ -1,4 +1,66 @@
...
+.orderListHeader,
+.orderHeader {
+    height: 78px;
+    display: flex;
+    align-items: center;
+    justify-content: stretch;
+    gap: 20px;
+}
+
+.orderListHeader > div,
```
Comment (olehrakwix) [inline app/routes/members-area.my-orders/route.module.scss:25]: these styles looks strange, can we add some class name? what is that? maybe columnTitle?

File: app/routes/members-area.my-orders/route.tsx [hunk 2]
```diff
@@ -1,10 +1,72 @@
...
+    const ordersResponse = await api.getOrders();
+
+    return { orders: ordersResponse.orders };
+};
+
 export default function MyOrdersPage() {
+    const { orders } = useLoaderData<typeof loader>();
+
+    const accordionItems = orders.map((order) => {
+        const formatDate = (date: Date) =>
```
Comment (olehrakwix) [inline app/routes/members-area.my-orders/route.tsx:21]: this might be moved out of `map` scope

File: app/routes/members-area.my-orders/route.tsx [hunk 3]
```diff
@@ -1,10 +1,72 @@
...
+                        <span className={styles.orderHeaderSectionName}>Order: </span>
+                        {order.number}
+                    </div>
+                    <div>
+                        <span className={styles.orderHeaderSectionName}>Status: </span>
+                        {order.status}
+                    </div>
+                    <div>
+                        <span className={styles.orderHeaderSectionName}>Total: </span>
+                        {order.priceSummary!.total!.formattedAmount}
```
Comment (olehrakwix) [inline app/routes/members-area.my-orders/route.tsx:45]: maybe better to use `?` in order to not crash if there will be some case that data is undefined?

File: lib/ecom/types.ts [hunk 4]
```diff
@@ -67,6 +67,10 @@ export type EcomApi = {
     getAllCategories: () => Promise<Collection[]>;
     getCategoryBySlug: (slug: string) => Promise<CollectionDetails | undefined>;
     getOrder: (id: string) => Promise<OrderDetails | undefined>;
+    getOrders: () => Promise<{
+        orders: OrderDetails[];
+        metadata?: orders.CursorPagingMetadata;
```
Comment (olehrakwix) [inline lib/ecom/types.ts:72]: why do we need this?

File: app/routes/members-area.my-orders/route.module.scss (hunk 1 above)
Comment (yurii-ve) [reply app/routes/members-area.my-orders/route.module.scss:25]: I didn't want to pollute jsx too much, as there 4 divs inside of a row.. but ok :)

File: lib/ecom/types.ts (hunk 4 above)
Comment (yurii-ve) [reply lib/ecom/types.ts:72]: it's something return by the SDK and contains information needed for the pagination (`count`, `hasNext`). I'll change it to return the same info as `queryProducts` does - `{items: [], totalCount: 0}`

**Review by olehrakwix — APPROVED** [review]

**Review by olehrakwix — APPROVED** [review]

**Pull request merged**